SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver_LoadBalancing.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialNetDriver.h"
+
+#include "Engine/NetworkObjectList.h"
+//
+#include "EngineClasses/SpatialActorChannel.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialSender.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "LoadBalancing/OwnershipLockingPolicy.h"
+#include "Schema/SpatialDebugging.h"
+#include "Utils/SpatialActorUtils.h"
+
+void USpatialNetDriver::GetLatestAuthorityChangeFromHierarchy(const AActor* HierarchyActor, uint64& OutTimestamp)
+{
+	if (HierarchyActor->GetIsReplicated())
+	{
+		if (USpatialActorChannel* Channel = GetOrCreateSpatialActorChannel(const_cast<AActor*>(HierarchyActor)))
+		{
+			if (Channel->GetAuthorityReceivedTimestamp() > OutTimestamp)
+			{
+				OutTimestamp = Channel->GetAuthorityReceivedTimestamp();
+			}
+		}
+	}
+
+	for (const AActor* Child : HierarchyActor->Children)
+	{
+		GetLatestAuthorityChangeFromHierarchy(Child, OutTimestamp);
+	}
+}
+
+void USpatialNetDriver::CollectActorsToMigrate(AActor* HierarchyActor, const AActor* OriginalActorBeingConsidered, VirtualWorkerId Destination, TArray<FMigrationInfo>& OutMigrationInfo, TSet<FNetworkObjectInfo*>& OutAdditionalConsider)
+{
+	if (HierarchyActor->GetIsReplicated()
+	&& HierarchyActor != OriginalActorBeingConsidered
+	&& HierarchyActor->HasAuthority())
+	{
+		if(FNetworkObjectInfo* Info = FindNetworkObjectInfo(HierarchyActor))
+		{
+			OutAdditionalConsider.Add(Info);
+			FMigrationInfo MigInfo = {HierarchyActor, Destination};
+			OutMigrationInfo.Add(MigInfo);
+		}
+	}
+
+	for (AActor* Child : HierarchyActor->Children)
+	{
+		CollectActorsToMigrate(Child, OriginalActorBeingConsidered, Destination, OutMigrationInfo, OutAdditionalConsider);
+	}
+}
+
+void USpatialNetDriver::ServerReplicateActors_HandleLoadBalancing(TArray<FNetworkObjectInfo*>& ConsiderList, TArray<FMigrationInfo>& OutMigrationInfo)
```

Comment:
Both `EntityCreationRateLimit` and `ActorReplicationRateLimit` limit how many actors are actually processed in this frame. `ActorReplicationRateLimit` probably shouldn't be used in general, and we should remove it, but I know NWX make use of the former.